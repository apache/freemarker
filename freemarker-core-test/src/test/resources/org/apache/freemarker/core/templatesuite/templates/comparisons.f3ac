<#--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->
<html>
<head>
<title>FreeMarker: Numeric Operations Test</title>
</head>
<body>

<p>A simple test follows:</p>

<p>${message}</p>

<p>Start with the increment operator:</p>
<#assign a1 = 0>
<p>a1 = ${a1}</p>
<#assign a1 = a1 + 1>
<p>a1 = ${a1}</p>
<#assign a1 = a1 + 1>
<p>a1 = ${a1}</p>

<p>Now the decrement operator:</p>
<#assign a2 = 5>
<p>a2 = ${a2}</p>
<#assign a2 = a2 - 1>
<p>a2 = ${a2}</p>
<#assign a2 = a2 - 1>
<p>a2 = ${a2}</p>
<#assign a2 = a2 - 1>
<p>a2 = ${a2}</p>
<#assign a2 = a2 - 1>
<p>a2 = ${a2}</p>
<#assign a2 = a2 - 1>
<p>a2 = ${a2}</p>
<#assign a2 = a2 - 1>
<p>a2 = ${a2}</p>
<#assign a2 = a2 - 1>

<p>Now the add operator:</p>
<#assign op1 = 5>
<#assign op2 = 3>
<#assign op3 = op1 + op2>
<p>op1 = ${op1}, op2 = ${op2}, op3 = ${op3}</p>
<#assign op3 = op3 + op2>
<p>op3 = ${op3}</p>

<p>And the subtract operator:</p>
<#assign op3 = op1 - op2 >
<p>op1 = ${op1}, op2 = ${op2}, op3 = ${op3}</p>
<#assign op3 = op3 - op2 >
<p>op3 = ${op3}</p>

<p>The comparison operators:</p>
<#assign list1 = [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 ]>
<#list list1 as item>
   <p>Item is: ${item}</p>
   <#if item lt 5>
   <p>Item is less than five.</p>
   </#if>
   <#if item <= 7>
   <p>Item is less than or equals to seven.</p>
   </#if>
   <#if item gt 2>
   <p>Item is greater than two.</p>
   </#if>
   <#if (item >= 10)>
   <p>Item is greater than or equal to ten.</p>
   </#if>
</#list>

<#-- Signum-based optimization test, all 9 permutations: -->
<#-- 1 -->
<@assert !(0 != 0) />
<@assert (0 == 0) />
<@assert !(0 > 0) />
<@assert (0 >= 0) />
<@assert !(0 < 0) />
<@assert (0 <= 0) />
<#-- 2 -->
<@assert !(3 != 3) />
<@assert (3 == 3) />
<@assert !(3 > 3) />
<@assert (3 >= 3) />
<@assert !(3 < 3) />
<@assert (3 <= 3) />
<#-- 3 -->
<@assert !(-3 != -3) />
<@assert (-3 == -3) />
<@assert !(-3 > -3) />
<@assert (-3 >= -3) />
<@assert !(-3 < -3) />
<@assert (-3 <= -3) />
<#-- 4 -->
<@assert (3 != 0) />
<@assert !(3 == 0) />
<@assert (3 > 0) />
<@assert (3 >= 0) />
<@assert !(3 < 0) />
<@assert !(3 <= 0) />
<#-- 5 -->
<@assert (0 != 3) />
<@assert !(0 == 3) />
<@assert !(0 > 3) />
<@assert !(0 >= 3) />
<@assert (0 < 3) />
<@assert (0 <= 3) />
<#-- 6 -->
<@assert (-3 != 0) />
<@assert !(-3 == 0) />
<@assert !(-3 > 0) />
<@assert !(-3 >= 0) />
<@assert (-3 < 0) />
<@assert (-3 <= 0) />
<#-- 7 -->
<@assert (0 != -3) />
<@assert !(0 == -3) />
<@assert (0 > -3) />
<@assert (0 >= -3) />
<@assert !(0 < -3) />
<@assert !(0 <= -3) />
<#-- 8 -->
<@assert (-3 != 3) />
<@assert !(-3 == 3) />
<@assert !(-3 > 3) />
<@assert !(-3 >= 3) />
<@assert (-3 < 3) />
<@assert (-3 <= 3) />
<#-- 9 -->
<@assert (3 != -3) />
<@assert !(3 == -3) />
<@assert (3 > -3) />
<@assert (3 >= -3) />
<@assert !(3 < -3) />
<@assert !(3 <= -3) />
<#-- Again, now on runtime: -->
<#assign m3 = -3>
<#assign p3 = 3>
<#assign z = 0>
<#-- 1 -->
<@assert !(z != z) />
<@assert (z == z) />
<@assert !(z > z) />
<@assert (z >= z) />
<@assert !(z < z) />
<@assert (z <= z) />
<#-- 2 -->
<@assert !(p3 != p3) />
<@assert (p3 == p3) />
<@assert !(p3 > p3) />
<@assert (p3 >= p3) />
<@assert !(p3 < p3) />
<@assert (p3 <= p3) />
<#-- 3 -->
<@assert !(m3 != m3) />
<@assert (m3 == m3) />
<@assert !(m3 > m3) />
<@assert (m3 >= m3) />
<@assert !(m3 < m3) />
<@assert (m3 <= m3) />
<#-- 4 -->
<@assert (p3 != z) />
<@assert !(p3 == z) />
<@assert (p3 > z) />
<@assert (p3 >= z) />
<@assert !(p3 < z) />
<@assert !(p3 <= z) />
<#-- 5 -->
<@assert (z != p3) />
<@assert !(z == p3) />
<@assert !(z > p3) />
<@assert !(z >= p3) />
<@assert (z < p3) />
<@assert (z <= p3) />
<#-- 6 -->
<@assert (m3 != z) />
<@assert !(m3 == z) />
<@assert !(m3 > z) />
<@assert !(m3 >= z) />
<@assert (m3 < z) />
<@assert (m3 <= z) />
<#-- 7 -->
<@assert (z != m3) />
<@assert !(z == m3) />
<@assert (z > m3) />
<@assert (z >= m3) />
<@assert !(z < m3) />
<@assert !(z <= m3) />
<#-- 8 -->
<@assert (m3 != p3) />
<@assert !(m3 == p3) />
<@assert !(m3 > p3) />
<@assert !(m3 >= p3) />
<@assert (m3 < p3) />
<@assert (m3 <= p3) />
<#-- 9 -->
<@assert (p3 != m3) />
<@assert !(p3 == m3) />
<@assert (p3 > m3) />
<@assert (p3 >= m3) />
<@assert !(p3 < m3) />
<@assert !(p3 <= m3) />
</body>
</html>
