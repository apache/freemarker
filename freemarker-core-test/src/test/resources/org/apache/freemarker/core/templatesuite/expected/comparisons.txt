/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
<html>
<head>
<title>FreeMarker: Numeric Operations Test</title>
</head>
<body>

<p>A simple test follows:</p>

<p>Hello, world!</p>

<p>Start with the increment operator:</p>
<p>a1 = 0</p>
<p>a1 = 1</p>
<p>a1 = 2</p>

<p>Now the decrement operator:</p>
<p>a2 = 5</p>
<p>a2 = 4</p>
<p>a2 = 3</p>
<p>a2 = 2</p>
<p>a2 = 1</p>
<p>a2 = 0</p>
<p>a2 = -1</p>

<p>Now the add operator:</p>
<p>op1 = 5, op2 = 3, op3 = 8</p>
<p>op3 = 11</p>

<p>And the subtract operator:</p>
<p>op1 = 5, op2 = 3, op3 = 2</p>
<p>op3 = -1</p>

<p>The comparison operators:</p>
   <p>Item is: 0</p>
   <p>Item is less than five.</p>
   <p>Item is less than or equals to seven.</p>
   <p>Item is: 1</p>
   <p>Item is less than five.</p>
   <p>Item is less than or equals to seven.</p>
   <p>Item is: 2</p>
   <p>Item is less than five.</p>
   <p>Item is less than or equals to seven.</p>
   <p>Item is: 3</p>
   <p>Item is less than five.</p>
   <p>Item is less than or equals to seven.</p>
   <p>Item is greater than two.</p>
   <p>Item is: 4</p>
   <p>Item is less than five.</p>
   <p>Item is less than or equals to seven.</p>
   <p>Item is greater than two.</p>
   <p>Item is: 5</p>
   <p>Item is less than or equals to seven.</p>
   <p>Item is greater than two.</p>
   <p>Item is: 6</p>
   <p>Item is less than or equals to seven.</p>
   <p>Item is greater than two.</p>
   <p>Item is: 7</p>
   <p>Item is less than or equals to seven.</p>
   <p>Item is greater than two.</p>
   <p>Item is: 8</p>
   <p>Item is greater than two.</p>
   <p>Item is: 9</p>
   <p>Item is greater than two.</p>
   <p>Item is: 10</p>
   <p>Item is greater than two.</p>
   <p>Item is greater than or equal to ten.</p>
   <p>Item is: 11</p>
   <p>Item is greater than two.</p>
   <p>Item is greater than or equal to ten.</p>
   <p>Item is: 12</p>
   <p>Item is greater than two.</p>
   <p>Item is greater than or equal to ten.</p>

</body>
</html>
